pkg/controller/v1beta1/inferenceservice/reconcilers/deployment/deployment_reconciler.go

```diff
@@ -0,0 +1,199 @@
+		Name:      r.Deployment.Name,
+	}, existingDeployment)
+	if err != nil {
+		if apierr.IsNotFound(err) {
+			return constants.CheckResultCreate, nil
+		}
+		return constants.CheckResultUnknown, err
+	}
+	//existed, check equalment
+	if semanticDeploymentEquals(r.Deployment, existingDeployment) {
+		return constants.CheckResultExisted, nil
+	}
+	return constants.CheckResultUpdate, nil
+}
+
+func semanticDeploymentEquals(desired, existing *appsv1.Deployment) bool {
+	return equality.Semantic.DeepEqual(desired.Spec, existing.Spec)
+}
+
+func setDefaultPodSpec(podSpec *corev1.PodSpec) {
+	if podSpec.DNSPolicy == "" {
+		podSpec.DNSPolicy = corev1.DNSClusterFirst
+	}
+	if podSpec.RestartPolicy == "" {
+		podSpec.RestartPolicy = corev1.RestartPolicyAlways
+	}
+	if podSpec.TerminationGracePeriodSeconds == nil {
+		TerminationGracePeriodSeconds := int64(corev1.DefaultTerminationGracePeriodSeconds)
+		podSpec.TerminationGracePeriodSeconds = &TerminationGracePeriodSeconds
+	}
+	if podSpec.SecurityContext == nil {
+		podSpec.SecurityContext = &corev1.PodSecurityContext{}
+	}
+	if podSpec.SchedulerName == "" {
+		podSpec.SchedulerName = corev1.DefaultSchedulerName
+	}
+	for i := range podSpec.Containers {
+		container := &podSpec.Containers[i]
+		if container.TerminationMessagePath == "" {
+			container.TerminationMessagePath = "/dev/termination-log"
+		}
+		if container.TerminationMessagePolicy == "" {
+			container.TerminationMessagePolicy = corev1.TerminationMessageReadFile
+		}
+		if container.ImagePullPolicy == "" {
+			container.ImagePullPolicy = corev1.PullIfNotPresent
+		}
+	}
+}
+
+func setDefaultDeploymentSpec(spec *appsv1.DeploymentSpec) {
+	if spec.Strategy.Type == "" {
+		spec.Strategy.Type = appsv1.RollingUpdateDeploymentStrategyType
+	}
+	if spec.Strategy.RollingUpdate == nil {
+		spec.Strategy.RollingUpdate = &appsv1.RollingUpdateDeployment{
+			MaxUnavailable: &intstr.IntOrString{Type: intstr.String, StrVal: "25%"},
+			MaxSurge:       &intstr.IntOrString{Type: intstr.String, StrVal: "25%"},
+		}
+	}
+	if spec.RevisionHistoryLimit == nil {
+		revisionHistoryLimit := int32(10)
+		spec.RevisionHistoryLimit = &revisionHistoryLimit
+	}
+	if spec.ProgressDeadlineSeconds == nil {
+		progressDeadlineSeconds := int32(600)
+		spec.ProgressDeadlineSeconds = &progressDeadlineSeconds
+	}
+}
+
+//Reconcile ...
+func (r *DeploymentReconciler) Reconcile() (*appsv1.Deployment, error) {
+	//reconcile Deployment
+	deployment := &appsv1.Deployment{}
+	checkResult, err := r.checkDeploymentExist(r.client)
+	if err != nil {
+		return nil, err
+	}
+	log.Info("deployment reconcile", "checkResult", checkResult, "err", err)
+	if checkResult == constants.CheckResultCreate {
+		err = r.client.Create(context.TODO(), r.Deployment)
+	} else if checkResult == constants.CheckResultUpdate { //CheckResultUpdate
+		err = r.client.Update(context.TODO(), r.Deployment)
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	err = r.client.Get(context.TODO(), types.NamespacedName{
```

Comment:
We can return the deployment in `checkDeploymentExist` so we do not have to call twice

Reply:
Got it